### PR TITLE
executor: Chunk size control test for Limit+TableScan and Limit+IndexScan

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Add some unit tests to prove that chunk size control works in these cases.
Following up #9512, this PR is a subtask of #9166.

### What is changed and how it works?
1. Use `mocktikv.Cluster` to construct some specified cases of data distribution.
2. Introduce `testSlowClient` to mock delay on some specified regions.
3. Add unit tests for `Limit+TableScan` and `Limit+IndexScan`.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: executor/chunk_size_control_test.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
+package executor_test
```
Comment (lamxTyler) [inline executor/chunk_size_control_test.go]: Miss license.

Comment (qw4990) [reply executor/chunk_size_control_test.go]: updated

File: executor/chunk_size_control_test.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
+// Copyright 2015 PingCAP, Inc.
```
Comment (lamxTyler) [inline executor/chunk_size_control_test.go]: 2019

Comment (qw4990) [reply executor/chunk_size_control_test.go]: My fault. 😟
Fixed now.

**Pull request merged**